src/main/java/frc/robot/subsystems/swerve/Swerve.java

```diff
@@ -286,4 +300,48 @@ public boolean isAtAngle(Rotation2d targetAngle) {
         return isAtAngle && isStopping;
     }
 
+    private static ChassisSpeeds applyDeadbandSpeeds(ChassisSpeeds chassisSpeeds) {
+        double newXSpeed = chassisSpeeds.vxMetersPerSecond;
+        if (Math.abs(chassisSpeeds.vxMetersPerSecond) <= SwerveConstants.DRIVE_NEUTRAL_DEADBAND) {
+            newXSpeed = 0;
+        }
+
+        double newYSpeed = chassisSpeeds.vyMetersPerSecond;
+        if (Math.abs(chassisSpeeds.vyMetersPerSecond) <= SwerveConstants.DRIVE_NEUTRAL_DEADBAND) {
+            newYSpeed = 0;
+        }
+
+        double newOmegaSpeed = chassisSpeeds.omegaRadiansPerSecond;
+        if (Math.abs(chassisSpeeds.omegaRadiansPerSecond) <= SwerveConstants.ROTATION_NEUTRAL_DEADBAND.getRadians()) {
+            newOmegaSpeed = 0;
+        }
+
+        return new ChassisSpeeds(newXSpeed, newYSpeed, newOmegaSpeed);
+    }
+
+    private static ChassisSpeeds fieldRelativeToRobotRelativeSpeeds(ChassisSpeeds fieldRelativeSpeeds, Rotation2d allianceRelativeAngle) {
+        return ChassisSpeeds.fromFieldRelativeSpeeds(fieldRelativeSpeeds, allianceRelativeAngle);
+    }
+
+    private static ChassisSpeeds powersToSpeeds(double xPower, double yPower, double thetaPower, DriveSpeed driveSpeed, SwerveConstants constants) {
+        return new ChassisSpeeds(
+                xPower * driveSpeed.translationSpeedFactor * constants.velocityAt12VoltsMetersPerSecond(),
+                yPower * driveSpeed.translationSpeedFactor * constants.velocityAt12VoltsMetersPerSecond(),
+                thetaPower * driveSpeed.rotationSpeedFactor * constants.maxRotationalVelocityPerSecond().getRadians()
+        );
+    }
+
+    private static ChassisSpeeds discretize(ChassisSpeeds chassisSpeeds) {
+        return ChassisSpeeds.discretize(chassisSpeeds, CycleTimeUtils.getCurrentCycleTime());
+    }
+
+    private static boolean isStill(ChassisSpeeds chassisSpeeds) {
+        return Math.abs(chassisSpeeds.vxMetersPerSecond) <= SwerveConstants.DRIVE_NEUTRAL_DEADBAND
+                && Math.abs(chassisSpeeds.vyMetersPerSecond) <= SwerveConstants.DRIVE_NEUTRAL_DEADBAND
+                && Math.abs(chassisSpeeds.omegaRadiansPerSecond) <= SwerveConstants.ROTATION_NEUTRAL_DEADBAND.getRadians();
+    }
+
+    public void applyStateHandler(SwerveStateHelper swerveStateHelper){
+        this.stateHandler = swerveStateHelper;
+    }
```

Comment:
```suggestion
    public void setStateHandler(SwerveStateHelper swerveStateHelper){
        this.stateHandler = swerveStateHelper;
    }
```
and move this to the beginning of the class where all the setter

Reply:
also do this in robot, and make the swerve and poseestimator private final (not static)